src/hostapi/wasapi/pa_win_wasapi.c

```diff
@@ -6152,7 +6228,9 @@ PA_THREAD_FUNC ProcThreadPoll(void *param)
         return (UINT32)paUnanticipatedHostError;
 
     // Request fine (1 ms) granularity of the system timer functions to guarantee correct logic around WaitForSingleObject
-    SystemTimer_SetGranularity(&timer, 1);
+    // Note: while granularity is already set by OpenStream() we still must be sure that user code did not
+    // change granularity between Pa_OpenStream() and Pa_StartStream()
+    SystemTimer_SetGranularity(&stream->timer, 1);
```

Comment:
Doesn't this mean you are calling `timeBeginPeriod` twice, but `timeEndPeriod` only once?

> You must match each call to timeBeginPeriod with a call to timeEndPeriod, specifying the same minimum resolution in both calls

https://learn.microsoft.com/en-us/windows/win32/api/timeapi/nf-timeapi-timebeginperiod

Reply:
No, `SystemTimer_RestoreGranularity` is called in the end of each thread function/worker and also in `CloseStream` therefore all calls to `SystemTimer_SetGranularity` match `SystemTimer_RestoreGranularity`.

Reply:
Ok, I see that the timer Set/Restore calls are matched.